drivers/sensor/max17055/max17055.c

```diff
@@ -205,14 +250,100 @@ static int max17055_sample_fetch(const struct device *dev,
 	return 0;
 }
 
+static int max17055_exit_hibernate(struct max17055_data *priv)
+{
+	LOG_DBG("Exit hibernate");
+
+	if (max17055_reg_write(priv, SOFT_WAKEUP, SOFT_WAKEUP_WAKEUP)) {
+		return -EIO;
+	}
+	if (max17055_reg_write(priv, HIB_CFG, HIB_CFG_CLEAR)) {
+		return -EIO;
+	}
+	if (max17055_reg_write(priv, SOFT_WAKEUP, SOFT_WAKEUP_CLEAR)) {
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int max17055_write_config(struct max17055_data *priv,
+				 const struct max17055_config *const config)
+{
+	uint16_t design_capacity = capacity_to_max17055(config->rsense_mohms,
+							config->design_capacity);
+	uint16_t d_qacc = design_capacity / 32;
+	uint16_t d_pacc = d_qacc * 44138 / design_capacity;
+	uint16_t i_chg_term = current_ma_to_max17055(config->rsense_mohms, config->i_chg_term);
+	uint16_t v_empty = voltage_mV_to_max17055(config->v_empty);
+
+	LOG_DBG("Writing configuration parameters");
+	LOG_DBG("DesignCap: %u, dQAcc: %u, IChgTerm: %u, VEmpty: %u, dPAcc: %u",
+		design_capacity, d_qacc, i_chg_term, v_empty, d_pacc);
+
+	if (max17055_reg_write(priv, DESIGN_CAP, design_capacity)) {
+		return -EIO;
+	}
+	if (max17055_reg_write(priv, D_QACC, d_qacc)) {
+		return -EIO;
+	}
+	if (max17055_reg_write(priv, ICHG_TERM, i_chg_term)) {
+		return -EIO;
+	}
+	if (max17055_reg_write(priv, V_EMPTY, v_empty)) {
+		return -EIO;
+	}
+	if (max17055_reg_write(priv, D_PACC, d_pacc)) {
+		return -EIO;
+	}
+	if (max17055_reg_write(priv, MODEL_CFG, MODELCFG_REFRESH)) {
+		return -EIO;
+	}
+
+	uint16_t model_cfg = MODELCFG_REFRESH;
+
+	while (model_cfg & MODELCFG_REFRESH) {
+		max17055_reg_read(priv, MODEL_CFG, &model_cfg);
+		k_sleep(K_MSEC(10));
+	}
+
+	return 0;
+}
+
+static int max17055_init_config(struct max17055_data *priv,
+				const struct max17055_config *const config)
+{
+	int16_t hib_cfg;
+
+	if (max17055_reg_read(priv, HIB_CFG, &hib_cfg)) {
+		return -EIO;
+	}
+
+	if (max17055_exit_hibernate(priv)) {
+		return -EIO;
+	}
+
+	if (max17055_write_config(priv, config)) {
+		return -EIO;
+	}
+
+	if (max17055_reg_write(priv, HIB_CFG, hib_cfg)) {
+		return -EIO;
+	}
+
+	return 0;
+}
+
```

Comment:
Is this logically part of same commit?

Reply:
To me, this is the smallest unit of useful change:

In order to correctly set the parameters we're defining in the device tree, we must properly follow the initialisation guide. I guess we could separate some of this out into a separate commit that takes the chip out of hibernate and does nothing, but that feels somewhat useless.

Reply:
looks right to me